code/modules/mob/living/carbon/human/death.dm

```diff
@@ -16,6 +16,19 @@
 /mob/living/carbon/human/death(gibbed)
 	if(stat == DEAD)
 		return
+
+	if(reagents && reagents.has_reagent("hearty_punch"))
+		src << "<span class='info'><b>A blast of energy revitalizes you. You won't die that easily!</b></span>"
+		var/amount_of_punch = reagents.get_reagent_amount("hearty_punch")
+		adjustBruteLoss(-amount_of_punch)
+		adjustFireLoss(-amount_of_punch)
+		adjustToxLoss(-amount_of_punch)
+		adjustOxyLoss(-amount_of_punch)
+		modify_drunkenness(80) //There has to be SOME downside
+		reagents.remove_reagent("hearty_punch", amount_of_punch)
+		reagents.add_reagent("atropine", amount_of_punch) //To ensure that they don't immediately bleed out

```

Comment:
Hearty Punch is tedious to make and requires basically a full chem dispenser's worth of energy to make even ten units. I just wanted to make a nonconventional cocktail with an actual effect - not to mention if someone is bludgeoning you, they can likely kill you faster than you can heal.